HydroShare Export Share Modal

rajadain opened the pull request:
## Overview

Changes the share modal of a project to match the new mockups. They HydroShare sharing part will come in a forthcoming card.

Connects #2511 

### Demo

![2017-11-30 09 59 45](https://user-images.githubusercontent.com/1430060/33437326-91d49062-d5b5-11e7-87f6-c77a3ea2c91d.gif)

### Notes

The Scenario sharing uses the same modal as before. Should this also be updated? I'm creating a card for follow up discussion: #2538.

The HydroShare parts of the mockup haven't been implemented yet, they will come in a follow up card: #2514.

## Testing Instructions

* Check out this branch and `./scripts/bundle.sh --debug`
* Go to [:8000/](http://localhost:8000/) and sign in.
* Create or open a TR-55 project. In the project menu, select "Share".
* Turn on link sharing for that project. Copy the link and open in a private browser window (where you're not signed in). Ensure the project is viewable.
* Turn off link sharing. Refresh the private browser window. Ensure you get a 404.
* Try turning link sharing on and off a bunch of times. Ensure the copy button works in all cases.

rajadain commented:
Also tagging @jfrankl for visual review.

jfrankl commented:
Looks good. One thing: I think the "Make private" button should go away. That's being replaced with this idea of a sharable link. If a shareable link is enabled, the project is effectively public. If it's not, it's effectively private.

rajadain commented:
There is some language in the "Make Private" / "Make Public" modals:

![image](https://user-images.githubusercontent.com/1430060/33443244-20188176-d5c5-11e7-8852-8456eeded254.png)

![image](https://user-images.githubusercontent.com/1430060/33443259-286bc55e-d5c5-11e7-9eea-5d1093095e73.png)

Should we replicate any of it in the new MultiShare modal?

caseycesari commented:
We may be able to condense it to something like "Anyone with the link will be able to view this project", and show that near the link sharing button.

jfrankl commented:
Something like this?

![image](https://user-images.githubusercontent.com/1809908/33445377-aee0f816-d5ca-11e7-82b7-ecead94ef684.png)

I like Casey's language above better than what's in this mockup. @ajrobbins curious what you think.

ajrobbins commented:
I like the new language better too!

rajadain commented:
Removed the Make Private / Make Public button, added language, added the new MultiShare modal to the Project Listing page as well.

![image](https://user-images.githubusercontent.com/1430060/33447433-491f477e-d5d1-11e7-9abc-aa7af5c50aa7.png)

rajadain commented:
Going to rebase and merge. Thanks for the reviews everyone!
